### ISSUE TITLE
airbase-ng -D|--clients <filename> not working

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#739]
**Reported by olivervbk on 12 May 2010 20:10 UTC**

Running
```
#airbase-ng -e test -f disallow -D mac_file mon0
```
does not affect behavior(with or without '-f disallow')

Example max_file:
```
00:11:22:33:44:55\n
00:00:00:00:00:00\n
```

While
```
#airbase-ng -e test -f disallow -d 00:11:22:33:44:55 mon0
```
works fine.

I've been tinkering with the source code (airbase-ng.c) and apparently the pMAC_t pMAC list generated by addMACfile() ( calling addMAC()) is not generated correctly(zero items returned by getMACcount() ). 

Checked on revision 1696

**Comment by aircrack-ng** [comment]
**Comment by olivervbk on 12 May 2010 20:10 UTC**

I've uploaded a fix.

The problem was that addMACfile() passed a string with a trailing '\n' to the getmac() function.

This fix should be temporary since it doesn't take account osx or windows formatting of text files('\r' and '\r\n').

**Comment by aircrack-ng** [comment]
**Comment by misterx on 12 May 2010 20:10 UTC**

Thanks for the fix but I'll probably fix it in getmac() so that in case we have such thing elsewhere, it doesn't happen again.

**Comment by aircrack-ng** [comment]
**Modified by misterx on 12 May 2010 20:10 UTC**

**Comment by aircrack-ng** [comment]
**Comment by misterx on 12 May 2010 20:10 UTC**

Fixed in aircrack-ng/aircrack-ng@29bd0fe6755fe5432fc2122700667e2918fa46bf. Thanks for pinpointing the issue.
